app/src/main/java/org/mozilla/focus/session/SessionManager.java

```diff
@@ -71,6 +71,11 @@ private void createSessionFromIntent(Context context, SafeIntent intent) {
         final String action = intent.getAction();
 
         if (Intent.ACTION_VIEW.equals(action)) {
+            final String dataString = intent.getDataString();
+            if (TextUtils.isEmpty(dataString)) {
```

Comment:
This works but I wonder: do we want to catch the error closer to where it's used? e.g. we can create a session with a static method that can return null in situations like this. The upside is that we're less likely to regress if the code changes but there are big downsides:
1. it's harder to take action for specific failures in the static method both because you only get "null" as a return value and because it's far removed from where the data is initially introduced
1. The static method would require a lot of refactoring given there are multiple constructors

Then again, our current set-up is pretty good at finding regressions too (catch at the source, throw for illegal state, add a test).

Just thinking aloud...